### PR TITLE
Change 2FA activities to more generic security activities

### LINKED ISSUE #4747: Hide two-factor auth filter in activities
* disable "Two-factor authentication" if none of those apps are enabled 
* see nextcloud/activity#150

cc @ChristophWurst

### PR DESCRIPTION
**Pull request by ChristophWurst**
Fixes #4747.

The new filter is called "Security" as suggested by @LukasReschke in https://github.com/nextcloud/server/issues/4747#issuecomment-299998772.

TODO:
- [x] adjust backup codes app
- [x] adjust TOTP app https://github.com/nextcloud/twofactor_totp/pull/175
- [x] adjust U2F app https://github.com/nextcloud/twofactor_u2f/pull/53

### CODE REVIEW

File: settings/Activity/SecurityFilter.php [hunk 1]
```diff
@@ -44,19 +44,19 @@ public function allowedApps() {
 	}
 
 	public function filterTypes(array $types) {
-		return array_intersect(['twofactor'], $types);
+		return array_intersect(['security'], $types);
```
Comment (nickvergessen) [inline settings/Activity/SecurityFilter.php]: I would add the `'personal_settings'` which shows password and email changes as well

Comment (nickvergessen) [inline settings/Activity/SecurityFilter.php]: Or we move those to the security as well for now.

Comment (ChristophWurst) [inline settings/Activity/SecurityFilter.php]: > I would add the 'personal_settings' which shows password and email changes as well

Are those activities we already have or something to be added later on?

Comment (nickvergessen) [inline settings/Activity/SecurityFilter.php]: They exist

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

File: settings/Activity/SecurityFilter.php (hunk 1 above)
Comment (ChristophWurst) [inline settings/Activity/SecurityFilter.php]: Alright. Then let's move the activities from https://github.com/nextcloud/server/pull/4330 to this new filter. But aren't we too close to a release to do so?

Comment (nickvergessen) [inline settings/Activity/SecurityFilter.php]: Well it's deleting one file and replacing one string, I'd say it's okay?

**Comment by nickvergessen** [comment]
`Test\Authentication\TwoFactorAuth\ManagerTest` fails

**Review by LukasReschke — APPROVED** [review]

**Pull request merged**